src/llmq/quorums_blockprocessor.cpp

```diff
@@ -298,14 +298,14 @@ bool CQuorumBlockProcessor::IsCommitmentRequired(Consensus::LLMQType llmqType, c
     return isMiningPhase && !hasMinedCommitment && !allowMissingQc;
 }
 
-// WARNING: This method returns uint256() on the first block of the DKG interval (because the block hash is not known yet)
+// This method returns uint256() for blocks prior to the DKG interval
 uint256 CQuorumBlockProcessor::GetQuorumBlockHash(Consensus::LLMQType llmqType, const CBlockIndex* pindex)
 {
     auto& params = Params().GetConsensus().llmqs.at(llmqType);
 
     int nHeight = pindex->nHeight;
     int quorumStartHeight = nHeight - (nHeight % params.dkgInterval);
-    if (pindex->nHeight <= quorumStartHeight + 1) {
+    if (pindex->nHeight < quorumStartHeight) {
```

Comment:
That whole check can vanish now if I didn't miss something. `nHeight` will now always be at or above the `quorumStartHeight` due to the calculation above. The comment on top of this method can also vanish now.